sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -288,6 +266,16 @@ public void AddLink(string traceparent, string? tracestate, IDictionary<string,
                 _currentActivity = StartActivitySourceActivity();
                 if (_currentActivity != null)
                 {
+                    _isAllDataRequested = _currentActivity.GetIsAllDataRequestedFlag();
```

Comment:
This class variable is not needed. We can just use the return of GetIsAllDataRequestedFlag directly.